code/modules/atmospherics/machinery/portable/pump.dm

```diff
@@ -58,14 +57,16 @@
 	excited = TRUE
 
 	var/turf/local_turf = get_turf(src)
+
 	var/datum/gas_mixture/sending
 	var/datum/gas_mixture/receiving
-	if(direction == PUMP_OUT) // Hook up the internal pump.
-		sending = (holding ? holding.return_air() : air_contents)
-		receiving = (holding ? air_contents : local_turf.return_air())
+
+	if (holding) //Work with tank when inserted, otherwise - with area
+		sending = (direction == PUMP_IN ? holding.return_air() : air_contents)
+		receiving = (direction == PUMP_IN ? air_contents : holding.return_air())
 	else
-		sending = (holding ? air_contents : local_turf.return_air())
-		receiving = (holding ? holding.return_air() : air_contents)
+		sending = (direction == PUMP_IN ? local_turf.return_air() : air_contents)
+		receiving = (direction == PUMP_IN ? air_contents : local_turf.return_air())
```

Comment:
why not keep the old pattern for this? don't like it?

Reply:
Also, previously "IN" during the pump-tank mode meant tank, now "IN" always means pump. This makes the logic easier to understand than before when direction flipped around when you inserted the tank.